blockstore/src/main/java/brooklyn/location/blockstore/azure/arm/AzureArmVolumeManager.java

```diff
@@ -0,0 +1,411 @@
+    
+    // Azure "unmanaged disks" don't support separate creation and attachment phases.
+    // Therefore we override methods like createAttachAndMountVolume. If/when we switch
+    // to using azure's DiskApi then we can refactor this code again. However, that is
+    // not yet supported in jclouds (v2.0.1).
+    
+    // TODO:
+    //  - What should the "lun" number be? Make this configurable?
+    //  - waitForJobToComplete: make timeouts configurable?
+    //  - OSDeviceName: was "/dev/sdc" during testing; in contrast AWS has "/dev/xvdc".
+    //    Is this dependent on the image? Do we need to make the prefix configurable?
+
+    private static final Logger LOG = LoggerFactory.getLogger(AzureArmVolumeManager.class);
+
+    @SuppressWarnings("unused")
+    private static final String PROVIDER = "azurecompute-arm";
+    
+    private static final String DEVICE_PREFIX = "/dev/sd";
+    private static final String OS_DEVICE_PREFIX = "/dev/sd";
+
+    private static final Duration TIMEOUT = Duration.minutes(2);
+    
+    @Override
+    public MountedBlockDevice createAttachAndMountVolume(JcloudsMachineLocation machine, BlockDeviceOptions deviceOptions,
+            FilesystemOptions filesystemOptions) {
+        AttachedBlockDevice attached = createAndAttachBlockDevice(machine, deviceOptions);
+        createFilesystem(attached, filesystemOptions);
+        return mountFilesystem(attached, filesystemOptions);
+    }
+
+    @Override
+    public void deleteBlockDevice(BlockDevice blockDevice) {
+        LOG.info("Deleting device: {}", blockDevice);
+        
+        AzureArmBlockDevice azureBlockDevice = (AzureArmBlockDevice) blockDevice;
+
+        String region = getRegionName(blockDevice.getLocation());
+        String resourceGroupName = azureBlockDevice.getResourceGroupName();
+        String storageAccountName = azureBlockDevice.getStorageAccountName();
+        AzureComputeApi api = getApi(blockDevice.getLocation());
+        deleteStorageAccount(api, resourceGroupName, storageAccountName, region);
+    }
+    
+    @Override
+    public MountedBlockDevice attachAndMountVolume(JcloudsMachineLocation machine, BlockDevice device,
+            BlockDeviceOptions options, FilesystemOptions filesystemOptions) {
+        throw new UnsupportedOperationException("Cannot attach pre-existing block-device using 'unmanaged disk' api: machine="+machine+"; device="+device);
+    }
+
+    @Override
+    public BlockDevice unmountFilesystemAndDetachVolume(MountedBlockDevice device) {
+        throw new UnsupportedOperationException("Cannot detach block-device using 'unmanaged disk' api: device="+device);
+    }
+
+    @Override
+    public BlockDevice createBlockDevice(JcloudsLocation location, BlockDeviceOptions options) {
+        throw new UnsupportedOperationException("Cannot create block-device using 'unmanaged disk' api, without attaching to VM: location="+location+"; options="+options);
```

Comment:
@bostko can't do that with jclouds azurecompute-arm yet. The block device is only actually created when you call `vmApi.create()`, which does a PUT to update the VM. All the code before that point is just building up the parameters to that API call.

If/when we have the "managed disks" api support in jclouds, then we can do as you suggest (to create the disk, and then to attach it to the vm).